src/blocks/HighlightBlock/elements/Component.tsx

```diff
@@ -0,0 +1,135 @@
+    header: 'max-h-8',
+    card: 'h-48',
+  },
+  large: {
+    overallSize: 'w-1/2',
+    header: 'h-full',
+    card: 'min-h-72',
+  },
+  xl: {
+    overallSize: 'w-2/3',
+    header: 'h-full',
+    card: 'min-h-96',
+  },
+}
+
+export const HighlightElementsComponent: React.FC<Props> = ({
+  title,
+  array,
+  size,
+  cardsToShow,
+}) => {
+  const selectedSize = sizes[size]
+  const [api, setApi] = React.useState<CarouselApi>()
+  const [current, setCurrent] = React.useState(0)
+
+  React.useEffect(() => {
+    if (!api) {
+      return
+    }
+    setCurrent(api.selectedScrollSnap())
+    api.on('select', () => {
+      setCurrent(api.selectedScrollSnap())
+    })
+  }, [api])
+
+  let cardsToDisplay = 'md:basis-1/3 lg:basis-1/3'
+  if (cardsToShow === 'two') {
+    cardsToDisplay = 'md:basis-1/2 lg:basis-1/2'
+  }
+
+  return (
+    <div className="flex flex-col justify-center items-center w-full p-4">
+      <H3 className="mb-2">{title}</H3>
+      <Carousel
+        opts={{
+          loop: true,
+          align: 'start',
+        }}
+        setApi={setApi}
+        className={`${selectedSize?.overallSize}`} // this changes the size of the entire component
+      >
+        <CarouselContent>
+          {array.map((element, index) => (
+            <CarouselItem key={index} className={`w-full ${cardsToDisplay} rounded-lg`}>
+              <Card
+                className={`${selectedSize?.card} bg-cover bg-center relative w-full`}
+                style={{ backgroundImage: `url(${element.image})` }}
+              >
+                <CardContent className="absolute inset-0 flex flex-col bg-neutral-950 bg-opacity-70 text-white items-start justify-between p-4 rounded-lg">
+                  <H3 className="line-clamp-4">{element.name}</H3>
+                  {(size == 'large' || size == 'xl') && element.summary && (
+                    <CardDescription className="text-white line-clamp-3">
+                      {element.summary}
+                    </CardDescription>
+                  )}
+                  <Button asChild className="text-white p-0" variant={'link'}>
+                    <Link target="_blank" href={element.link} className="text-black">
+                      More Details
+                    </Link>
+                  </Button>
+                </CardContent>
+                <CardFooter></CardFooter>
+              </Card>
+            </CarouselItem>
+          ))}
+        </CarouselContent>
+        <div className="flex justify-start items-start gap-3 pt-2">
+          <Button
+            className="rounded-full w-10 p-1"
+            variant="outline"
+            onClick={() => api?.scrollTo(current - 1)}
+          >
+            <ArrowLeft />
+          </Button>
+          <Button
+            className="rounded-full w-10 p-1"
+            variant={'outline'}
+            onClick={() => api?.scrollTo(current + 1)}
+          >
+            <ArrowRight />
+          </Button>
+        </div>
```

Comment:
```suggestion
        <CarouselPrevious />
        <CarouselNext />
```
We can use the predefined components here

Reply:
yeah, sure but it gives a very standard feeling, doesn't it? I honestly liked the buttons at the bottom way more..

Reply:
Are you suggesting using the pre-built buttons on the sides of the cards? It gives me a very "standard" feeling...

Reply:
yes. I think it's actually a strength in this case to be "standard" — users are already familiar with how carousels typically behave, so sticking to recognizable patterns helps with usability.

Also, using the built-in `CarouselPrevious` and `CarouselNext` components from shadcn gives us a few advantages:
– Better accessibility out of the box (like proper tab order and keyboard navigation)
– Cleaner code, since we don’t have to manage state or reimplement the same logic

And if we want a different look, we can still style them with Tailwind to match our needs — without losing the built-in benefits.